src/shared/components/Preview/Preview.tsx

```diff
@@ -0,0 +1,212 @@
+import * as React from 'react';
+import {
+  Resource,
+  NexusClient,
+  GetFileOptions,
+  NexusFile,
+} from '@bbp/nexus-sdk';
+import { Button, Collapse, Table } from 'antd';
+import PDFViewer from './PDFPreview';
+import useNotification from '../../hooks/useNotification';
+
+const parseResourceId = (url: string) => {
+  const fileUrlPattern = /files\/([\w-]+)\/([\w-]+)\/(.*)/;
+  if (fileUrlPattern.test(url)) {
+    const [, , , resourceId] = url.match(fileUrlPattern) as string[];
+    return decodeURIComponent(resourceId);
+  }
+
+  if (/[w-]+/.test(url)) {
+    return url;
+  }
+
+  return '';
+};
+
+const parseProjectUrl = (projectUrl: string) => {
+  const projectUrlR = /projects\/([\w-]+)\/([\w-]+)\/?$/;
+  const [, org, proj] = projectUrl.match(projectUrlR) as string[];
+  return [org, proj];
+};
+
+const Preview: React.FC<{
+  resource: Resource;
+  nexus: NexusClient;
+}> = ({ resource, nexus }) => {
+  const notification = useNotification();
+  const [previewAsset, setPreviewAsset] = React.useState<any | undefined>();
+  const previewDivRef = React.useRef<HTMLDivElement>(null);
+  const [orgLabel, projectLabel] = parseProjectUrl(resource._project);
+
+  /* close preview when anywhere outside the preview is clicked */
+  React.useEffect(() => {
+    document.addEventListener('mousedown', e => {
```

Comment:
is it possible to use antd Modal window here? it has all the styles and built in functionality like click outside to close etc

Reply:
I did try with the antd modal but it was introducing styling challenges. Have left as is and fixed the clean up issue picked up by deepscan

Reply:
👍